compiler/rustc_span/src/symbol.rs

```diff
@@ -3009,6 +3034,25 @@ impl Ident {
         self.name.can_be_raw() && self.is_reserved()
     }
 
+    pub fn is_raw_lifetime_guess(self) -> bool {
+        // this should be kept consistent with `Parser::expect_lifetime` found under
+        // compiler/rustc_parse/src/parser/ty.rs
+        let name_without_apostrophe = self.without_first_quote();
+        name_without_apostrophe.name != self.name
+            && ![kw::UnderscoreLifetime, kw::StaticLifetime].contains(&self.name)
+            && name_without_apostrophe.is_raw_guess()
```

Comment:
Since `rustc_span` actually defines `Ident` and `is_reserved`, you could reasonably also define a canonical "`is_reserved_lifetime`" which you can then reuse.